### DISCUSSION TITLE
Why is a very large number of requests being made

### DISCUSSION THREAD

**Discussion by deus-developer** [BeanieODM/beanie#533] [Question]
I have very simple models where there may be several nested links, but I don't understand why the request turns into an aggregation of 100 elements in a pipeline. Moreover, any work with links creates a catastrophic number of queries in the database and discourages any desire to use them. Maybe in the next update it can be somehow optimized?

**Comment by roman-right** [comment]
Hi. Could you please be more specific? Please provide:
- The models and the aggregation pipeline, that looks too heavy. 
- Cases, when the number of queries is getting too big. With code examples to reproduce the problem on my side.
